### PR TITLE
Support for auto-instrumentation

### PR DESCRIPTION
**Pull request by labbati**
### Description

This PR adds support for autoinstrumenting PHP code.

### Readiness checklist
- [x] [Changelog entry](docs/changelog.md) added, if necessary
- [x] Tests added for this feature/bug

### CODE REVIEW

File: src/DDTrace/Integrations/ZendFramework/ZendFrameworkIntegration.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * Loads version 1 of zend framework.
+     */
+    private static function loadV1()
+    {
+        dd_trace('Zend_Application', 'setOptions', function () {
+            $args = func_get_args();
+            $options = $args[0];
+
+            $classExist = class_exists('DDTrace_Ddtrace');
```
Comment (pawelchcki) [inline src/DDTrace/Integrations/ZendFramework/ZendFrameworkIntegration.php]: :eye_twitch: at the random casing - I assume its as per Zend recommendation ?

File: src/ext/ddtrace.c [hunk 2]
```diff
@@ -203,6 +203,51 @@ static PHP_FUNCTION(dd_trace) {
     RETURN_BOOL(rv);
 }
 
+// This function allows untracing a function.
+static PHP_FUNCTION(dd_untrace) {
```
Comment (pawelchcki) [inline src/ext/ddtrace.c]: Good stuff !!!!

File: tests/Frameworks/Laravel/Version_4_2/composer.json [hunk 3]
```diff
@@ -16,8 +16,7 @@
 			"app/database/seeds"
 		],
 		"psr-4": {
-			"App\\": "app/",
-			"DDTrace\\": "../../../../src/DDTrace/"
```
Comment (pawelchcki) [inline tests/Frameworks/Laravel/Version_4_2/composer.json]: 🎉

File: tests/ext/reset_function_tracing.phpt [hunk 4]
```diff
@@ -0,0 +1,32 @@
+--TEST--
+Check a function can be untraced.
```
Comment (pawelchcki) [inline tests/ext/reset_function_tracing.phpt]: Small nitpick - this test could also verify other functions/methods still remain traced after one is untraced.

File: .circleci/config.yml [hunk 5]
```diff
@@ -51,6 +51,8 @@ aliases:
     image: php:7.2
     name: request_replayer
     command: sh -c "curl --output index.php https://raw.githubusercontent.com/DataDog/dd-trace-php/dev/tests/request_replayer/index.php -q && php -S 0.0.0.0:80 index.php"
+    environment:
+      DD_REQUEST_DUMPER_FILE: dump.json
```
[the remark this answers is not in the document]
Comment (labbati) [reply .circleci/config.yml]: 😄

File: src/DDTrace/Integrations/ZendFramework/ZendFrameworkIntegration.php (hunk 1 above)
Comment (labbati) [reply src/DDTrace/Integrations/ZendFramework/ZendFrameworkIntegration.php]: correct! it actually took as a while to understand! 😄

**Comment by SammyK** [comment]
Amazing work @labbati! :)

**Pull request merged**